source/opt/folding_rules.cpp

```diff
@@ -344,6 +544,15 @@ spvtools::opt::FoldingRules::FoldingRules() {
   rules_[SpvOpPhi].push_back(RedundantPhi());
 
   rules_[SpvOpSelect].push_back(RedundantSelect());
+
+  rules_[SpvOpFAdd].push_back(RedundantFAdd());
+  rules_[SpvOpFSub].push_back(RedundantFSub());
+  rules_[SpvOpFMul].push_back(RedundantFMul());
+  rules_[SpvOpFDiv].push_back(RedundantFDiv());
+
+  // TODO: We should probably add first class filtering for ExtInst
```

Comment:
I'll probably leave this as is for now. There are a few ExtInst ops that I think I'll want to constant-fold because our shaders use them (we aren't hitting *any* of these atm but I think that's because the shaders that use those with constant args rely on loop unrolling that we don't have active atm), but I don't anticipate many more non-const folding rules so probably this can be handled when the time comes.